src/cache/RedisQueryResultCache.ts

```diff
@@ -38,17 +44,28 @@ export class RedisQueryResultCache implements QueryResultCache {
     /**
      * Creates a connection with given cache provider.
      */
+
+
     async connect(): Promise<void> {
         const cacheOptions: any = this.connection.options.cache;
-        if (cacheOptions && cacheOptions.options) {
-            this.client = this.redis.createClient(cacheOptions.options);
-        } else {
-            this.client = this.redis.createClient();
+        if (this.clientType === "redis") {
+            if (cacheOptions && cacheOptions.options) {
+                this.client = this.redis.createClient(cacheOptions.options);
+            } else {
+                this.client = this.redis.createClient();
+            }
+        } else if (this.clientType === "ioredis") {
+            if (cacheOptions && cacheOptions.options) {this.client = this.redis.createClient();
```

Comment:
is it typo? this.client is being assigned two times?

Reply:
indeed it was. i have fixed it.